puttputt/logic.py

```diff
@@ -0,0 +1,223 @@
+    elif user_type == 'barista':
+        location = 'barista'
+        # if it's a sponsor -> redirect them to the sponsor page
+    elif user_type == 'sponsor':
+        location = 'sponsor'
+    else:
+        # if it's a manager -> redirect them to the manager page
+        location = 'manager'
+
+    return location
+
+def get_user_type(user):
+    user_type = 'player'
+
+    sponsor = get_sponsor(user)
+    manager = get_manager(user)
+    barista = get_barista(user)
+
+    if sponsor != None:
+        return 'sponsor'
+    elif manager != None:
+        return 'manager'
+    elif barista != None:
+        return 'barista'
+
+    return user_type
+
+def get_game_info(user):
+    print('getting game info for user...')
+
+    info = get_player(user)['player_info']
+
+    current_score = info.current_score
+    current_hole = info.current_hole
+
+    return {'current_score': current_score,'current_hole': current_hole}
+
+def add_stroke_to_player(player, stroke):
+    location = 'game'
+
+    profile = player['profile']
+    player = player['player_info']
+
+    player.current_hole == 0
+
+    player.current_hole += 1
+    player.current_score += int(stroke)
+
+    if player.current_hole > get_number_of_holes():
+        location = 'end_game'
+        tour = tournament_for_player(player)
+        leaderboard = LeaderBoard.objects.create(tournament=tour, profile=profile, score=player.current_score)
+        leaderboard.save()
+
+    player.save()
+
+    return location
+
+def get_number_of_holes():
+    return VenueInfo.objects.all()[:1].get().number_of_holes
+
+def pay_fee_for_sponsor(user):
+    sponsor = get_sponsor(user)
+    try: 
+        tournament = Tournament.objects.get(sponsor=sponsor)
+        tournament.fee_paid = True
+        tournament.save()
+    except:
+        return None
+
+def add_player_to_sponsor(user, player_username):
+    sponsor = get_sponsor(user)
+    tournament = Tournament.objects.get(sponsor=sponsor)
+    
+    player_user = User.objects.get(username=player_username)
+
+    player_profile = Profile.objects.get(user=player_user)
+
+    player_info = PlayerInfo.objects.get(profile=player_profile)
+
+    player_info.tournament = tournament
+    
+    player_info.save()
+
+def get_final_score(user):
+    player = get_player(user)['player_info']
+
+    return player.current_score
+
+def get_all_time_leaderboard():
+    return LeaderBoard.objects.all().order_by('score')[:50]
+
+def get_leaderboard_for(user):
+    tournament = get_player(user)['player_info'].tournament
+
+    return LeaderBoard.objects.filter(tournament=tournament).order_by('score')
+
+def check_game_complete_for(user):
+    complete = True
+
+    tournament = get_player(user)['player_info'].tournament
+
+    competitors = PlayerInfo.objects.filter(tournament=tournament)
+
+    for comp in competitors:
+        if comp.current_hole <= get_number_of_holes():
+            complete = False
+
+    return complete
+
+def set_user_type(to, username):
```

Comment:
The name "to" is a little weird. I guess it works though. You could change it to "user_type" or something.

Reply:
Yeah I didn't know whether or not to change those, it's a convention of Swift programming that I kinda carried over.

Reply:
I'll change it later

Reply:
I think it's fine. I'm going to mark this as resolved